homeassistant/components/wink.py

```diff
@@ -66,30 +85,109 @@
 ]
 
 
+def _write_config_file(file_path, config):
+    try:
+        with open(file_path, 'w') as conf_file:
+            conf_file.write(json.dumps(config, sort_keys=True, indent=4))
+    except IOError as error:
+        _LOGGER.error("Saving config file failed: %s", error)
+        raise IOError("Saving Wink config file failed")
+    return config
+
+
+def _read_config_file(file_path):
+    try:
+        with open(file_path, 'r') as conf_file:
+            return json.loads(conf_file.read())
+    except IOError as error:
+        _LOGGER.error("Reading config file failed: %s", error)
+        raise IOError("Reading Wink config file failed")
+
+
+def _request_app_setup(hass, config, config_path):
+    """Assist user with configuring the Wink dev application."""
+    configurator = get_component('configurator')
+
+    # pylint: disable=unused-argument
+    def wink_configuration_callback(callback_data):
+        """Handle configuration updates."""
+        _config_path = hass.config.path(WINK_CONFIG_FILE)
+        if os.path.isfile(_config_path):
+            config_file = _read_config_file(_config_path)
+            if config_file == DEFAULT_CONFIG:
+                error_msg = ("You didn't correctly modify wink.conf",
+                             " please try again")
+                configurator.notify_errors(_CONFIGURING['wink'], error_msg)
+            else:
+                setup(hass, config)
+        else:
+            # Shouldn't get here, but if we do, call setup again and create
+            #  default config again
+            setup(hass, config)
+
+    start_url = "{}{}".format(hass.config.api.base_url,
+                              WINK_AUTH_CALLBACK_PATH)
+
+    description = """If you haven't done so already.
+                     Please create a Wink developer app at
+                     https://developer.wink.com.
+                     Add a Redirect URI of {}.
+                     They will provide you a Client ID and secret
+                     after reviewing your request. (This can take several days)
+                     These need to be saved into the file located at: {}.
+                     (This file is hidden in Linux (prefixed with a '.'))
+                     Then come back here and hit the below button.
+                     """.format(start_url, config_path)
+
+    submit = "I have saved my Client ID and Client Secret into wink.conf."
```

Comment:
Why not have people enter it via the configurator UI ? You can add fields to `request_config`. Example: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/gpmdp.py#L98-L105